File: src/problems/dontmatter/113.cpp
int main(){
    int nOndas, inicio, clones, zetsus, totalN=0;
    bool perdeu = false;

    cin >> nOndas >> inicio;
    cin >> zetsus >> clones;

    clones += inicio;
    while (nOndas != 0){
        cin >> zetsus >> clones;
        totalN += clones;
        if (zetsus > totalN){
            perdeu = true;
        }
        totalN -= zetsus;
        nOndas--;
    }

    if (!perdeu){
        cout << "Naruto defendeu a vila";
    } else {
        cout << "Madara venceu";
    }
    return 0;
}

File: src/problems/dontmatter/113.py
ondas, clones_naruto = map(int, input().split())
defendeu_vila = True
for _ in range(ondas):
    zetsus, clones = map(int, input().split())
    clones_naruto += clones
    if zetsus > clones_naruto:
        defendeu_vila = False
    clones_naruto -= zetsus   

if defendeu_vila:
  print('Naruto defendeu a Vila')
else:
  print('Madara venceu')
